gate-saml/src/main/java/com/netflix/spinnaker/gate/security/saml/SecuritySamlProperties.java

```diff
@@ -159,4 +269,38 @@ public static class UserAttributeMapping {
     private String username;
     private String email;
   }
+
+  /// Filched DIRECTLY from spring-security private temporarily until all of this is replaced
+  // directly with the autowired spring configuration stuff
+  ///
+  // https://github.com/spring-projects/spring-security/blob/main/config/src/main/java/org/springframework/security/config/saml2/RelyingPartyRegistrationsBeanDefinitionParser.java
+  private static final ResourceLoader resourceLoader = new DefaultResourceLoader();
+
+  private static Saml2X509Credential getSaml2Credential(
+      String privateKeyLocation,
+      String certificateLocation,
+      Saml2X509Credential.Saml2X509CredentialType credentialType) {
+    RSAPrivateKey privateKey = readPrivateKey(privateKeyLocation);
+    X509Certificate certificate = readCertificate(certificateLocation);
+    return new Saml2X509Credential(privateKey, certificate, credentialType);
+  }
+
+  private static RSAPrivateKey readPrivateKey(String privateKeyLocation) {
+    Resource privateKey = resourceLoader.getResource(privateKeyLocation);
+    try (InputStream inputStream = privateKey.getInputStream()) {
+      return RsaKeyConverters.pkcs8().convert(inputStream);
+    } catch (Exception ex) {
+      throw new IllegalArgumentException(ex);
+    }
+  }
+
+  private static X509Certificate readCertificate(String certificateLocation) {
+    Resource certificate = resourceLoader.getResource(certificateLocation);
+    try (InputStream inputStream = certificate.getInputStream()) {
+      return (X509Certificate)
+          CertificateFactory.getInstance("X.509").generateCertificate(inputStream);
```

Comment:
Isn't there utility code in kork-crypto for this?

Reply:
PROBABLY? - but this is a direct copy/paste from Spring Security so was just re-using all of that Autowiring flow.  I'd like to replace all of this with the spring autowiring similar to the oauth2 autowiring changes I'm seeing.  BUT this is a "fix" vs. a... larger rewrite of that auth flow.